distributed/scheduler.py

```diff
@@ -5510,6 +5511,41 @@ def send_all(self, client_msgs: dict, worker_msgs: dict):
             except (CommClosedError, AttributeError):
                 self.loop.add_callback(self.remove_worker, address=worker)
 
+    def annotate_task(
+        self,
+        worker=None,
+        key: str = None,
+        annotations: Mapping = None,
+        resource_restrictions: Mapping = None,
+        annotate_dependents=True,
```

Comment:
I'm curious about the motivation behind this keyword.  Can you expand here?

Reply:
I am considering if an _annotator function_ should be able to enable/disable annotations of task dependents. Alternativevly, we can always apply annotations to task dependents.

Reply:
I agree that it's possibly in scope.  I'm curious if there are active use cases for it today that you have in mind.  If so, I would be curious to hear them (for example, maybe you need the results of gpu tasks are likely to also be gpu tasks).  If not, then I would recommend waiting on this until we know more.

My gut reaction is not to do this until we have a motivating use case.

Reply:
> for example, maybe you need the results of gpu tasks are likely to also be gpu tasks
Yes, to implement automatic use of a GPU ThreadPoolExecutor (#5084). If the initial array or dataframe creation is annotated with the GPU executor, all the following operations are as well. 

This will also be very helpful when mixing data types. By tracking the GPU use of each individual partition, we can utilize GPU and CPU workers simultaneously.